plugins/home/api-report.md

```diff
@@ -264,10 +294,40 @@ export type VisitListenerContextValue = {
 
 // @public
 export interface VisitsApi {
-  listUserVisits(queryParams?: VisitsApiQueryParams): Promise<Visit[]>;
-  saveVisit(saveParams: VisitsApiSaveParams): Promise<Visit>;
+  list(queryParams?: VisitsApiQueryParams): Promise<Visit[]>;
+  save(saveParams: VisitsApiSaveParams): Promise<Visit>;
+}
+
+// @public
+export class VisitsApiFactory implements VisitsApi {
+  protected constructor({
+    randomUUID,
+    limit,
+    retrieveAll,
+    persistAll,
+  }: VisitsApiFactoryOptions);
+  // (undocumented)
+  protected readonly limit: number;
+  // (undocumented)
+  list(queryParams?: VisitsApiQueryParams): Promise<Visit[]>;
+  // (undocumented)
+  protected persistAll: (visits: Array<Visit>) => Promise<void>;
```

Comment:
Drawback of using subclassing is that you expose the nature of the everything-at-once implementation details of your actual end classes. Maybe you wanted this exposed publicly to allow for reuse by others (the sort stuff etc), but arguably, it'd been good to have this entire class (and `VisitsApiFactoryOptions`) be internal and not visible outwards.

Reply:
It was my intention to expose `VisitsApiFactory` but I can see your point. I'm having some dificulties to put it into action. When I remove the export api-reports builds give me `// Warning: (ae-forgotten-export) The symbol "VisitsApiFactory" needs to be exported by the entry point index.d.ts`. It seems I'm unable to do it like this. Do you have any suggestion?

Reply:
@freben, based on yours and @camilaibs comments it seems to be solution for both points as I see them: A better way to extend your own VisitsApi implementations and relying only on storageApi.
I will follow it up on another PR to the feature branch.